samza-rest/src/main/java/org/apache/samza/rest/model/ApplicationInfo.java

```diff
@@ -0,0 +1,295 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.rest.model;
+
+import java.util.Objects;
+import org.codehaus.jackson.annotate.JsonProperty;
+
+
+public class ApplicationInfo {
+  @JsonProperty("finishTime")
+  private long _finishTime;
+
+  @JsonProperty("amContainerLogs")
+  private String _amContainerLogs;
+
+  @JsonProperty("trackingUI")
+  private String _trackingHistory;
+
+  @JsonProperty("state")
+  private String _state;
+
+  @JsonProperty("user")
+  private String _user;
+
+  @JsonProperty("id")
+  private String _id;
+
+  @JsonProperty("clusterId")
+  private String _clusterId;
+
+  @JsonProperty("finalStatus")
+  private String _finalStatus;
+
+  @JsonProperty("amHostHttpAddress")
+  private String _amHostHttpAddress;
+
+  @JsonProperty("applicationType")
+  private String _applicationType;
+
+  @JsonProperty("applicationTags")
+  private String _applicationTags;
+
+  @JsonProperty("finishedTime")
+  private String _finishedTime;
+
+  @JsonProperty("preemptedResourceMB")
+  private String _preemptedResourceMB;
+
+  @JsonProperty("preemptedResourceVCores")
+  private String _preemptedResourceVCores;
+
+  @JsonProperty("numNonAMContainerPreempted")
+  private String _numNonAMContainerPreempted;
+
+  @JsonProperty("numAMContainerPreempted")
+  private String _numAMContainerPreempted;
+
+  @JsonProperty("progress")
+  private int _progress;
+
+  @JsonProperty("name")
+  private String _name;
+
+  @JsonProperty("startedTime")
+  private long _startedTime;
+
+  @JsonProperty("elapsedTime")
+  private long _elapsedTime;
+
+  @JsonProperty("diagnostics")
+  private String _diagnostics;
+
+  @JsonProperty("trackingUrl")
+  private String _trackingUrl;
+
+  @JsonProperty("queue")
+  private String _queue;
+
+  @JsonProperty("allocatedMB")
+  private int _allocatedMB;
+
+  @JsonProperty("allocatedVCores")
+  private int _allocatedVCores;
+
+  @JsonProperty("runningContainers")
+  private int _runningContainers;
+
+  @JsonProperty("memorySeconds")
+  private int _memorySeconds;
+
+  @JsonProperty("vcoreSeconds")
+  private int _vcoreSeconds;
+
+  ApplicationInfo() {
+
+  }
+
+  public ApplicationInfo(long finishTime, String amContainerLogs, String trackingHistory, String state, String user,
```

Comment:
Do we really care about all of these fields? Maybe only keep the ones we want and allow Jackson to ignore extra fields?

Reply:
Agreed. Will ignore fields that are not used.